sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -257,35 +257,27 @@ export class MessageSession extends LinkEntity<Receiver> {
     }
   }
 
-  protected createRheaLink(
+  protected async createRheaLink(
     options: ReceiverOptions,
     _abortSignal?: AbortSignalLike
   ): Promise<Receiver> {
-    return this._context.connection.createReceiver(options);
-  }
-
-  /**
-   * Creates a new AMQP receiver under a new AMQP session.
-   */
-  private async _init(
-    opts: { abortSignal?: AbortSignalLike; timeoutInMs?: number } = {}
-  ): Promise<void> {
-    try {
-      const sessionOptions = this._createMessageSessionOptions(this.identifier, opts.timeoutInMs);
-      await this.initLink(sessionOptions, opts.abortSignal);
-
-      if (this.link == null) {
-        throw new Error("INTERNAL ERROR: failed to create receiver but without an error.");
-      }
-
-      const receivedSessionId =
-        this.link.source &&
-        this.link.source.filter &&
-        this.link.source.filter[Constants.sessionFilterName];
+    this._lastSBError = undefined;
+    const link = await this._context.connection.createReceiver(options);
 
+    // When we ask for any sessions, but don't get one back, check whether
+    // service has sent any error.
+    if (
+      options.source &&
+      typeof options.source !== "string" &&
+      options.source.filter &&
+      Constants.sessionFilterName in options.source.filter &&
+      options.source.filter![Constants.sessionFilterName] === undefined
+    ) {
+      const receivedSessionId = link.source?.filter?.[Constants.sessionFilterName];
       let errorMessage: string = "";
-
       if (this._providedSessionId == null && receivedSessionId == null) {
+        await new Promise((resolve) => setTimeout(resolve, 1)); // yield to event-loop
```

Comment:
Another more complex way is to wait for error events.  However, from my testing and log, the error comes immediately after link is established and yielding looks good enough.